gunicorn/_reloader.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -
+#
+# This file is part of gunicorn released under the MIT license.
+# See the NOTICE for more information.
+
+import os.path
+import re
+import sys
+
+from watchdog.events import FileSystemEventHandler
+from watchdog.observers import Observer
+from watchdog.observers.polling import PollingObserver
+
+
+class Reloader(object):
+
+    def __init__(self, extra_files=None, interval=1, callback=None,
+                 engine='auto'):
+        super(Reloader, self).__init__()
+        if engine == 'auto' or engine == 'inotify':
+            self._observer = Observer()
+        else:
+            assert engine == 'poll'
+            self._observer = PollingObserver(timeout=interval)
+        self._observer.setDaemon(True)
+        self._handler = ReloadingEventHandler(callback)
+        self._dirs = set()
+
+        for filename in (extra_files or []):
```

Comment:
Just a nitpick, but adding an `if extra_files is not None:` branch might be more readable.